src/System.Configuration.ConfigurationManager/tests/System/Configuration/StringValidatorTests.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Configuration;
+using System.Reflection;
+using Xunit;
+
+namespace System.ConfigurationTests
+{
+    public class StringValidatorTests
+    {
+        [Fact]
+        public void Constructor_MinLength()
+        {
+            StringValidator validator = new StringValidator(5);
+            int expectedMinimumLength = 5;
+
+            int actualMinimumLength = Convert.ToInt32(typeof(StringValidator).GetField("_minLength", BindingFlags.Instance | BindingFlags.NonPublic | BindingFlags.GetField).GetValue(validator));
```

Comment:
We shouldn't be using reflection like this.  We should instead validate the behavior via the public surface area, e.g. construct the StringValidator with various values, and verify that Validate then does / does not throw based on the data being validated. (Same comment applies to other uses of reflection throughout.)